apis/python/src/tiledbsoma/common_nd_array.py

```diff
@@ -85,8 +89,16 @@ def _build_tiledb_schema(
     util.check_type("type", type, (pa.DataType,))
 
     # check on shape
-    if len(shape) == 0 or any(e <= 0 for e in shape):
-        raise ValueError("SOMA NDArray shape must be non-zero length tuple of ints > 0")
+    if is_sparse:
+        if len(shape) == 0 or any(e is not None and e <= 0 for e in shape):
+            raise ValueError(
+                "SOMASparseNDArray shape must be non-zero length tuple of positive ints or Nones"
```

Comment:
NIT: encoding subclass knowledge in the parent is slightly "ewwww" :-)   Possible fix: do the sparse-specific defaulting in a subclass specialization of this method, and then call the super()._build_tiledb_schema() method.

Reply:
put another way - the class-specific requirements should not be pushed up into this general-purpose superclass

Reply:
I think this might work best broken into two parts:

 1. Here, just validate non-emptiness:
    ```python
    if not shape:
      raise ValueError("ndarrays must have a nonzero number of dimensions")
    ```
 2. Validate the values when building dimensions below…

Reply:
done